react-carousel 2.0: Introduce carousel plugins (#586)
brainhubeu/react-carousel: docs/plugins/creatingCustomPlugins.md

          ...trackStyles,
          marginLeft: myCustomFuncToModifyMargin(),
        });
      }, [carouselProps.value]);
  },

  //  Here you can define what element will show up before and after 
  //  carousel items
  beforeCarouselItems: () => <div>humble navigation item</div>,
  afterCarouselItems: () => <div>humble navigation item</div>,

  // Define props which will be attached to .BrainhubCarousel__track
  carouselCustomProps: () => ({
    onMouseEnter: myCustomOnMouseEnterHandler,
    onMouseLeave: myCustomOnMouseLeaveHandler,
  }),

  // Modify react carousel events
  strategies: () => ({
      /**
       * Modify changing slide behavior
       * 
       * @param {number} original carousel value not modified by other plugins
       * @param {number} previous carousel value modified by other plugins
      **/
      [CAROUSEL_STRATEGIES.CHANGE_SLIDE]: (originalValue, previousValue) => {

        // modifies the value returned by the previous plugin by 
        // adding the value defined in the plugin options
        return previousValue + options.slidesToAdvance
      },
      [CAROUSEL_STRATEGIES.GET_TRANSFORM_OFFSET]: (originalValue, previousValue) => {

>         return previousValue + doSomeCalc();

Reviewer: I see `doSomeCalc` isn't defined in any place

Author: it's an example code. I believe users will know they will need to write their own functions to calculate new values

Reviewer: ok

### How to create react-carousel custom plugin

```js
// recoil is a lib for state management.
// More details here: https://recoiljs.org/
import { useRecoilState } from 'recoil';
import { 
  trackStylesState, 
  CAROUSEL_STRATEGIES,
} from '@brainhubeu/react-carousel';

/**

>  * This is custom plugin function which allow you to modify carousel look & behavior

Reviewer: ```suggestion
 * This is a custom plugin function which allows you to modify carousel look & behavior.
```

 *
 * @param { object } options defined by the carousel user
 * @param { object } carouselProps props defined for the react-carousel component
 * @param { node } ref.trackContainerRef track container red
 * @param { node } ref.carouselRef slide ref
 */
const myGreatPlugin = ({ options , carouselProps, ref }) => ({

  // Here you can apply hooks which can affect carousel behavior
  plugin: () => {
      const [trackStyles, setTrackStyles] = useRecoilState(trackStylesState);

> 

Reviewer: ```suggestion
```

      // modify trackStyle marginleft value when slide value has changed 
      // you can find more info regarding modifying carousel state here:

>       // https://beghp.github.io/gh-pages-rc-15/docs/plugins/accessingManagingState

Reviewer: ```suggestion
      // https://brainhubeu.github.io/react-carousel/docs/plugins/accessingManagingState
```
